### ISSUE TITLE
Can't sign in, Browser or app may not be secure

### ISSUE THREAD

**Issue by dancojocaru2000** [ytmdesktop/ytmdesktop#866]
### Consent

- [X] I verified that there is no open issue for the same subject.
- [X] I understand that **YTMDesktop have NO affiliation with Google or YouTube**

### Current Behavior

Google doesn't allow me to sign in:

![Couldn't sign you in Google page screenshot](https://user-images.githubusercontent.com/9626761/141196557-9b1d6984-1f6d-42d1-b560-8c055b1d4e89.png)

### Expected Behavior

Sign in.

### Steps To Reproduce

Download app, click sign in

### YTMDesktop

v1.13.0

### OS

Fedora

### OS Version

34

### Arch

x64

### Installation way

.AppImage

**Comment by Alipoodle** [comment]
On Windows machine here, but cannot seem to reproduce this.
We get around this issue with a different useragent than Electron Natively provides (to circumvent this)

Can I please confirm that you are indeed on 1.13?

**Comment by nikobonomi** [comment]
Same problem here,

the login is stuck on the 2-step verification phase

I'm on ubuntu 20.04, YTMDesktop installed trough snap

here is the snap info:
```
name:      youtube-music-desktop-app
summary:   YouTube Music Desktop
publisher: Adler (adlerluiz)
store-url: https://snapcraft.io/youtube-music-desktop-app
license:   unset
description: |
  Listen YouTube Music without need open the browser
commands:
  - youtube-music-desktop-app
snap-id:      S0db4gRY0r3Galb1B51Bt5h7zKbwqNj8
tracking:     latest/stable
refresh-date: today at 10:07 CET
channels:
  latest/stable:    1.13.0 2020-09-07 (6) 76MB -
  latest/candidate: ↑                          
  latest/beta:      ↑                          
  latest/edge:      ↑                          
installed:          1.13.0            (6) 76MB -
```


![Screenshot from 2021-11-15 10-13-52](https://user-images.githubusercontent.com/9111900/141754649-8f809e25-fde0-463d-9fcc-9a9d658e3e8b.png)

**Comment by BrOstro** [comment]
Getting the same issue as OP on Windows 10

**Comment by ausmreed** [comment]
For anyone having this issue, I followed these steps from another issue and was able to log in: 

1. Open up YTMD
2. Click "Sign in", nothing
3. Right-click system tray icon and select Guest
4. Second window opens up with login
5. Sign in via second window, still guest in main window
6. Close YTMD fully and reopen
7. Click "Sign in"
8. Sign in page opens in main window with user account logged out
9. Select account, normal sign in process
10. Signed in - Success

**Comment by BrOstro** [comment]
> For anyone having this issue, I followed these steps from another issue and was able to log in:
> 
> 1. Open up YTMD
> 2. Click "Sign in", nothing
> 3. Right-click system tray icon and select Guest
> 4. Second window opens up with login
> 5. Sign in via second window, still guest in main window
> 6. Close YTMD fully and reopen
> 7. Click "Sign in"
> 8. Sign in page opens in main window with user account logged out
> 9. Select account, normal sign in process
> 10. Signed in - Success

Thanks this worked for me

**Comment by jwhipp** [comment]
Just wanted to say I replicated this today with a fresh install of 1.13: 
![sign in blocked](https://user-images.githubusercontent.com/556332/142680172-797e1cda-c762-408b-970b-a83cfad8b536.png)

**Comment by DavidGoodwin** [comment]
My google account requires 2FA, and It seems to be stuck on a 'using your security key step'. 

I'm using YTMdesktop via an Ubuntu snap; perhaps the 2FA issue is down to the snap + needing access to a USB device?

**Comment by Alipoodle** [comment]
@DavidGoodwin There's a separate issue for that as that's an issue to do with support for the 2FA Keys in Electron.

The issue above is likely to do with the Faked Firefox useragent being considered out of date now.
I'm currently running a up to date Useragent and will verify if there are any issues with it before it get's pushed to Development and then soon a stable build.

**Comment by digihash** [comment]
> 

I can login on 

> For anyone having this issue, I followed these steps from another issue and was able to log in:
> 
> 1. Open up YTMD
> 2. Click "Sign in", nothing
> 3. Right-click system tray icon and select Guest
> 4. Second window opens up with login
> 5. Sign in via second window, still guest in main window
> 6. Close YTMD fully and reopen
> 7. Click "Sign in"
> 8. Sign in page opens in main window with user account logged out
> 9. Select account, normal sign in process
> 10. Signed in - Success

I can login in the guest part but even after I did that and restarted the program, I can't login in the normal part. It keeps complaining about the browser being outdated.

**Comment by TheArgentinian** [comment]
This worked but I had to try twice.

**Comment by nineinchnick** [comment]
I'm having the same issue on the latest MacOS (12). It worked with the login as guest workaround, but I had to click "Try Again" when logging in in the main window after restart. Weird.

**Comment by adkeyz** [comment]
> For anyone having this issue, I followed these steps from another issue and was able to log in:
> 
> 1. Open up YTMD
> 2. Click "Sign in", nothing
> 3. Right-click system tray icon and select Guest
> 4. Second window opens up with login
> 5. Sign in via second window, still guest in main window
> 6. Close YTMD fully and reopen
> 7. Click "Sign in"
> 8. Sign in page opens in main window with user account logged out
> 9. Select account, normal sign in process
> 10. Signed in - Success

Not working for me, tried it a few times, still can't sign in.

**Comment by jgspratt** [comment]
What does `Click "Sign in", nothing` mean?  Click `sign in` and then do nothing?  Or click `sign in` and have it fail after entering your email address?

**Comment by jgspratt** [comment]
I also am not clear on `Second window opens up with login` and then `Sign in via second window, still guest in main window`.  The window that opens up after selecting `guest` is the "second window" in my view (the first one was the one that opened originally, so `still guest in main window` doesn't make sense.

Do we sign in via the guest window ever in this procedure?

**Comment by jgspratt** [comment]
Ah, hang on, I guessed around and got it like this.

1. Open up YTMD
1. Click `Sign in` then **do nothing** (leave the email blank)
1. Right-click system tray icon and select `Guest`
1. Guest window opens.  **Ignore the guest window** and **go back to the original window** with the blank email address field (leave the guest window open--do NOT close it)
1. Enter your email address now (**in the non-guest window**).  Hit `next`.  Enter your password and login.
1. You now have two windows open: one fully signed in "main" window and one "guest" window.
1. Close YTMD fully (hit the `Quit youtube-music-desktop-app` menu item)
1. Re-open YTMD
1. You are already signed in - success

**Comment by DavidGoodwin** [comment]
@jgspratt unfortunately that doesn't work for me - on step 5 I get this after entering my email address.

![image](https://user-images.githubusercontent.com/203929/144464493-6409aae4-147f-43bf-84d8-bfddac85e44a.png)

**Comment by digihash** [comment]
I just downloaded the latest nightly build that was build 3 days ago and that one seems to work perfectly on my Windows 10 machine. See https://github.com/ytmdesktop/ytmdesktop/releases/tag/1.14.1

**Comment by DavidGoodwin** [comment]
Hm, I'm using an ubuntu snap, i suspect it's being difficult for other reasons.

**Comment by digihash** [comment]
> Hm, I'm using an ubuntu snap, I suspect it's being difficult for other reasons (v1.13.0) which may explain a lot.

Only stable releases seem to be pushed to Snap so you still use the stable release indeed. You can always build the source code yourself. Here is an interesting tutorial on how to do that on Ubuntu: https://www.howtogeek.com/105413/how-to-compile-and-install-from-source-on-ubuntu/

**Comment by jgspratt** [comment]
I installed mine via `brew install --cask ytmdesktop-youtube-music` which pulled form `https://github.com/ytmdesktop/ytmdesktop/releases/download/v1.13.0/YouTube-Music-Desktop-App-1.13.0.dmg` on MacOS `10.15.7 (19H1519)`

**Comment by eduardo-manrique-procodific** [comment]
It's still not working to me with the nightly version or running the latest version from de source code with `npm start`, i'll still getting same result

**Comment by bhhaskin** [comment]
sign in worked fine for me on macOS Big Sur v11.6. Installed via brew

**Comment by hendrimp1** [comment]
I am getting this error as well. Using nightly 1.14.1 on Windows 10 21H2.

**Comment by DavidGoodwin** [comment]
> > Hm, I'm using an ubuntu snap, I suspect it's being difficult for other reasons (v1.13.0) which may explain a lot.
> 
> Only stable releases seem to be pushed to Snap so you still use the stable release indeed. You can always build the source code yourself. Here is an interesting tutorial on how to do that on Ubuntu: https://www.howtogeek.com/105413/how-to-compile-and-install-from-source-on-ubuntu/

I realised I could just launch the appImage variant (from releases); that works fine - although I needed to login as guest, then restart the app, before i could login as myself.

**Comment by Alipoodle** [comment]
To update a bit.

1.14.1 is intended to improve it, we are using a new UserAgent within this which is Firefox 94.
HOWEVER This doesn't solve the issue. A few retries of it within the menu SHOULD display the normal sign in.

However this issue isn't going to be easy to solve really. 😅
We can't easily tell how Google detects if it's an insecure browser, and none of us have been able to reproduce it in a Development environment.
I caused it yesterday but a single retry and it was fine.

**Comment by Faek** [comment]
I was getting this issue where I couldn't log in when I downloaded version 1.13 and 1.14 and tried to do some of the guest account workarounds that were listed above in this thread, none of it worked for me.

However, here is what did allow me to work around the issue:

1. Uninstall any version of YT Music Desktop you currently have installed
2. Install Version 1.11 from releases: [1.11 release](https://github.com/ytmdesktop/ytmdesktop/releases/tag/v1.11.0)
3. Sign in to your account using version 1.11 (I had to complete a captcha in order to sign in to my account, but it worked fine)
4. Then download, install, and run either version 1.13 or 1.14 nightly and you will still be signed in from when you did it on v1.11

**Comment by JasonHorkles** [comment]
I managed to get it to work by dragging the sign in button to my browser then clicking the sign in button again in the app lol

**Comment by psulek** [comment]
> I am a new user and I was getting this issue where I couldn't log in when I downloaded version 1.13 and 1.14 and tried to do some of the guest account workarounds that were listed above in this thread, none of it worked for me.
> 
> However, here is what did allow me to work around the issue:
> 
> 1. Uninstall any version of YT Music Desktop you currently have installed
> 2. Install Version 1.11 from releases: [1.11 release](https://github.com/ytmdesktop/ytmdesktop/releases/tag/v1.11.0)
> 3. Sign in to your account using version 1.11 (I had to complete a captcha in order to sign in to my account, but it worked fine)
> 4. Then download, install, and run either version 1.13 or 1.14 nightly and you will still be signed in from when you did it on v1.11 and can use the latest version of the app

This WORKED for me. Installed version 1.11, sigin. Then download latest (1.13) extract content (via 7zip) of file `app-64.7z` (in exe setup) into installed location of app and everything works!

**Comment by gianjohansen** [comment]
Got it on a 1.14.1 fresh install, worked after the 6th or 7th retry. I didn't try anything to fix it, just randomly clicked the back button or the try again button.

**Comment by OlJohnny** [comment]
OS: Windows 10 - 21H1

Tried the method with the guest window above, didn't work (with 1.13 and 1.14.1, after multiple installation & uninstallations)

Clean Installing 1.11 logging in and installing 1.14.1 (without uninstalling 1.11) worked for me!

**Comment by psulek** [comment]
@OlJohnny  yep, as i wrote earlier ;-)

**Comment by Alipoodle** [comment]
Nightly 1.14.1 is actually based on 9ff1c160d2e005d51e2efb9a3492e6338f2ea32d (which is a few fixes I've added on Development branch)

But generally considered it's a "minor" fix...
Curious to how Google fully does it, and is likely to do with a few checks on the page to see if features exist and stuff. 🤷

**Comment by A-Matt** [comment]
The UA on this is allowed to be modified by the user in the config (in nightly) but there's some issues with modifying it sometimes.
*This user agent only applies upon accessing the login page*

That's why I went with a Firefox 94 as being modern update of the current one we use which was FF73
Might I ask which UA did you go with?

**Comment by marens101** [comment]
> Fresh install of 1.14.1 nightly on Windows 11 21H2/22000.348 (verified the appdata roaming folder was removed)
> 
> Just kept clicking "Try Again" in the on the google login prompt, 6-7 times after it logged me in. It's (likely) the Google login page failing open vs closed (my theory anyways).
> 
> **My theory:** There's probably some kind of fail open trigger that temp disables the "out of date browser control" on the account level if you trigger it too many times.
> 
> TL;DR -- just keep clicking "Try Again" on the google login prompt until it works.
> 
> Citation:
> 
> * I'm pentester and sometimes I just keep clicking next a bunch of times to see what happens

This one worked for me. 

My guess is guest signin works for most people since that seems to use a different (older) login form, but it doesn't support physical 2fa keys. The default one, after getting past this hurdle, does seem to support them. I'm using a fresh install of 1.13, hadn't gotten to the nightly fixes yet.

**Comment by eccentrichead** [comment]
> Fresh install of 1.14.1 nightly on Windows 11 21H2/22000.348 (verified the appdata roaming folder was removed)
> 
> Just kept clicking "Try Again" in the on the google login prompt, 6-7 times after it logged me in. It's (likely) the Google login page failing open vs closed (my theory anyways).
> 
> **My theory:** There's probably some kind of fail open trigger that temp disables the "out of date browser control" on the account level if you trigger it too many times.
> 
> TL;DR -- just keep clicking "Try Again" on the google login prompt until it works.
> 
> Citation:
> 
> * I'm pentester and sometimes I just keep clicking next a bunch of times to see what happens

This method appears to somewhat work with the `v1.13.0` stable release too, on my linuxmint 20.2 system. The step I followed was:
- Download the latest v1.14 nightly (appimage) and open the app.
- Click on the login section
- Clicked the `Try again` button a bunch of times until I got the enter password page, it then took me the approve app access via the 2FA auth section. From there the login was successful once approved the access.
- Login back using the v1.13 stable release. 

I think even without the nightly image this method should work with the stable release. I didn't test it though.

**Comment by jerbear2008** [comment]
> TL;DR -- just keep clicking "Try Again" on the google login prompt until it works.

This worked for me too!

**Comment by KimBum** [comment]
Had the same issue in current pre-release, mainly because I killed my session from google account session list.
Was able to sign in again immediately by opening the guest login window and trying again in original one.
Interesting thing that the guest sign in page seems to use a different login page (very different wording and styling)

**Comment by superbonaci** [comment]
> I managed to get it to work by dragging the sign in button to my browser then clicking the sign in button again in the app lol

works!

**Comment by christianopaets** [comment]
> I managed to get it to work by dragging the sign in button to my browser then clicking the sign in button again in the app lol

I don't know why, but this really works)))

**Comment by KimBum** [comment]
I am pretty sure, that dragging the button, which is basically a link onto the browser creates some kind of session that is shared with the Chromium inside Electron here. For Google that seems to verify it as a valid device/browser that is really used.

**Comment by timbuckto** [comment]
So I tried this 7 or 8 times in a row, and I followed the instructions to a T. I even tried setting up an App Password, but that didn't work either. 

What did work was to modify the `TransportSecurity` file. Changing the subdomain states showing `false`--> `true`. Make sure to make a backup copy of the file prior to editing the file. I would suggest these changes if the steps above don't work. Maybe this is just a bug in the `TransportSecurity` file that needs to be marked to true so that it can verify the session with [https://music.youtube.com/](https://music.youtube.com/) as the subdomain, and/or [https://account.google.com/](https://account.google.com/). Also, it looks like marking this as true _forces_ https... so this might be what is needed.

```
C:\Users\%Username%\AppData\Roaming\youtube-music-desktop-app
Filename: TransportSecurity
```

Example of the value after it's marked as true:
`"sts_include_subdomains":true`

Once changed you _should_ be able to login via the main window. I closed and re-opened the app just to make sure it stuck. And I was able to stream without issue.

**Note:** this file will constantly update and may change some of the values back to `false`... just leave it as it is, and you should be good to go.

**Comment by Alipoodle** [comment]
I'm locking this chat now as a lot of discussion has gone on, and there are many work arounds to this, but none of which are entirely fixing the issue at heart.

I will list a few links to suggested comments people can go to to attempt to fix things, however in summary the methods which people have seem to have gotten to work are:

- [Use Guest Mode, Restart and try again.](https://github.com/ytmdesktop/ytmdesktop/issues/866#issuecomment-974205806)
- [Using 1.14.1 / 1.11 (Strange that an older version would be better really?), Signing in and then reverting to a different version](https://github.com/ytmdesktop/ytmdesktop/issues/866#issuecomment-986094248)
- [Hit Try again a few times](https://github.com/ytmdesktop/ytmdesktop/issues/866#issuecomment-986908898)
    Some people get it on 1 attempt, some get it upwards of 8.
- [Drag Sign-in into a new Browser and Logging in there](https://github.com/ytmdesktop/ytmdesktop/issues/866#issuecomment-986114220)

**Comment by Alipoodle** [comment]
One comment I hadn't notice until now was the Different UI between Guest & Native application.
 This is likely due to the fact that we are using a Firefox 93 and the Electron version we are using doesn't support 100% of the features so it's going for an older thing.
 When using the Guest mode, it uses a UserAgent which is a 1-1 Chrome version to what the Chromium version the Electron is based on (something like 86, when latest is 96) (some features aren't implemented into Electron, such as Cast, but otherwise it should match nicely with Chrome equivalent )

That is likely why they differ. However I cannot guarantee 100% as this is Google Magic.

**Comment by Alipoodle** [comment]
Please note, that we are still looking into this.
I've been personally testing a few different methods which I believed to have got around this, but none of which have worked consistently enough.
User-agent seems to be one of the better working versions of this, but this hasn't been perfect and still triggers it every so often.

**Comment by Alipoodle** [comment]
Closing issue due to no longer supportinng YouTube Music Desktop v1 + Release of YouTube Music v2.
Please re-open this issue if this is still occuring in the latest build of the application, however as a re-write has been done on the entire project most issues will be fixed, no longer a feature or re-worked.

This issue should be resolved within YTMDv2, I suspect that the main problem here was to do with the Chrome version being used as you hould be able to get the "v3" Login screen now over the v2 which YTMDv1 was giving before.

Some bits like Security Keys aren't fully supported for a SECONDARY factor but work as a primary factor for logging in.
